lib/Extractor/KLEEBuilder.cpp

```diff
@@ -169,6 +172,23 @@ ref<Expr> ExprBuilder::makeSizedArrayRead(unsigned Width, StringRef Name,
                                                                                   klee::ConstantExpr::create(i, Width))));
       PowerTwoBitsMap[Origin] = PowerExpr;
     }
+    if (Origin->NumSignBits > 1) {
+      ref<Expr> MSB = ExtractExpr::create(Var, Width-1, Expr::Bool);
+      ref<Expr> SignBitsExpr = klee::ConstantExpr::create(1, Expr::Bool);
+      for (int I=Width-1; I >= (int) (Width-Origin->NumSignBits); --I)
+        SignBitsExpr = AndExpr::create(SignBitsExpr, EqExpr::create(MSB, ExtractExpr::create(Var, I, Expr::Bool)));
+      if (Origin->NumSignBits != Width) {
```

Comment:
These extra cases to handle the number of sign bits are added to find more optimizations. The case of (NumSignBits != width) here makes sure that if we have 'N' sign bits, then (N+!)th bit from MSB will be opposite to the sign bit value.
For the cases when, (NumSignbits == Width), such cases don't need to be applied and that's why I didn't add anything.